### PR TITLE
loosen date-time validation

### PR DESCRIPTION
**Pull request by ankhyk**
should not report error when date-time value is just missing a Z in the end

### CODE REVIEW

File: lib/swaggerValidator/ajvSchemaValidator.ts [hunk 1]
```diff
@@ -229,6 +229,23 @@ const shouldSkipError = (error: ErrorObject, cxt: SchemaValidateContext) => {
     return true;
   }
 
+  // If a response has property which x-ms-secret value is "true" in post we can skip this error
```
Comment (raych1) [inline lib/swaggerValidator/ajvSchemaValidator.ts]: Comment needs to be changed.

File: lib/swaggerValidator/ajvSchemaValidator.ts [hunk 2]
```diff
@@ -229,6 +229,23 @@ const shouldSkipError = (error: ErrorObject, cxt: SchemaValidateContext) => {
...
+  if (
+    cxt.isResponse &&
+    keyword === "format" &&
+    schema === "date-time" &&
```
Comment (raych1) [inline lib/swaggerValidator/ajvSchemaValidator.ts]: Can you confirm if schema will have the value of 'date-time-rfc1123'?

File: test/modelValidation/swaggers/specification/dateTime/dateTime.json [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+          },
+          "type": "string",
+          "description": "State of user session."
+        },
+        "activeDirectoryUserName": {
+          "type": "string",
+          "description": "The active directory user name."
+        },
+        "createTime": {
+          "format": "date-time",
```
Comment (raych1) [inline test/modelValidation/swaggers/specification/dateTime/dateTime.json]: Can you have another case of 'date-time-rfc1123'?

File: lib/swaggerValidator/ajvSchemaValidator.ts [hunk 4]
```diff
@@ -229,6 +229,26 @@ const shouldSkipError = (error: ErrorObject, cxt: SchemaValidateContext) => {
     return true;
   }
 
+  // If payload has property with date-time parameter and its value is valid except missing "Z" in the end we can skip this error
+  if (
+    cxt.isResponse &&
```
Comment (raych1) [inline lib/swaggerValidator/ajvSchemaValidator.ts]: `cxt.isResponse` can also be removed coz the check applies for request too.

**Review by raych1 — COMMENTED** [review]
Can you modify the swagger and payload to cover `datetime` property validation in request either?

**Comment by ankhyk** [comment]
> Can you modify the swagger and payload to cover `datetime` property validation in request either?

The payload already has date-time property in both request and response

**Review by raych1 — APPROVED** [review]

**Pull request merged**